### ISSUE TITLE
Identifying and accommodating changes by MLIR converter 

### ISSUE THREAD

**Issue by keithm-xmos** [xmos/ai_tools#319]
The new MLIR based TF → TFLite converter may produce some unexpected behavior, and also introduces new features.

### Batchnorm and activation fusing

We should check if the new converter can fuse conv-activation-BN or only conv-BN-activation.

**Remove** unnecessary RESHAPE ops or optimize buffers

Since we assume the tensor sizes are not dynamic, RESHAPE shouldn’t do anything. The TFLite converter sometimes inserts these before/after certain Keras layers (e.g. global average pooling and fully connected). A pass should remove RESHAPE ops where appropriate, or ensure that it is effectively a noop. We already have some support for this, but we should evaluate where we are and how much it needs to be extended.

### Channel-wise quantization

This has been expected to be introduced at some point. Data tensors may be quantized channel-wise (i.e. different zero_point/scale for each channel, rather than one for the whole tensor). We need to think about how this will affect us.

The following xformer passes will be affected:

All conv2d passes:

- zero_point_bias calculations: these have been implemented with this potential change in mind
- ‘pad’ custom option: this is closely tied to how padding is applied in the kernels

Fully connected replacement pass:

- zero_point_bias calculations: this has been implemented with this potential change in mind
- small effect on output tensor mutation (to INT16) 

ReplaceXCOREWeightBiasOperatorPass:

- implicitly affects all operators with weights/biases
- unified bias calculation: this has been implemented with this potential change in mind

All LUT activation passes:

- these should probably be disabled for channel-wise padded tensors (the number of LUTs needed would require too much memory).

Padding passes:

- only affected by changes to the ‘pad’ custom option

Global average pool replacement pass:

- only the bias/scale/shift calculation and tensor is affected

**The following operators' integration will be affected:**

All conv2d operators:

- ‘pad’ custom option: this is closely tied to how padding is applied in the kernels

Global average pool:

- The bias/scale/shift tensor will now be channel-wise

**The following kernels will be affected:**

- TODO
